### ISSUE TITLE
Assert may have no effect for ModelicaTest.Media.TestAllProperties.PartialMediumFunctions

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#1529]
**Reported by otter on 27 Jun 2014 14:50 UTC**
The test models for media under ModelicaTest.Media.TestAllProperties all inherit from ModelicaTest.Media.TestAllProperties.PartialMediumFunctions where the base functionality is implemented. Especially, there are the following statements:
```
  // When iterating at the initial time the asserts below could be violated.
  // To avoid an error, the check is only performed at the end of the simulation.
  when terminal() then
    assert(err_T <= eps, "Error: abs(medium.T - T) > eps ...");
  end when;
``` 
As a result, the assert(..) is triggered at the terminal time point. In Dymola 2014 FD01 and 2015, the assert is correctly triggered and is shown in the output window. However, the simulation returns wrongly with "successful simulation" (function simulateModel returns with true). In certain circumstances this gives the wrong impression that the experiment was successful.

The assert should be defined so that it appears shortly after initialization, e.g.:
```
  // When iterating at the initial time the asserts below could be violated.
  // To avoid an error, the check is only performed shortly after initialization.
  if time > t_min then
    assert(err_T <= eps, "Error: abs(medium.T - T) > eps ...");
  end if;
  ...
```

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/1529

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 27 Jun 2014 14:55 UTC**
Fixed in ac09096b2d4c25e776a9ca8f603b958a4522db50 (trunk) and in 1958abcfb413427dacc6bec545c360663060dd2d (maintenance)

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 27 Jun 2014 15:27 UTC**
The issue is also present in ModelicaTest.Media.TestAllProperties.FlueGasSixComponents. 
Fixed in 03fb3adeb698bb76b3d2637bf961040317ebb1a7 (trunk) and in a03276811cdf18332ebbab0b3849ad391c352f29 (maintenance)
